plugins/woocommerce-docs/src/Manifest/ManifestProcessor.php

```diff
@@ -15,17 +17,20 @@ class ManifestProcessor {
 	 * @param int    $logger_action_id The logger action ID.
 	 */
 	public static function process_manifest( $manifest, $logger_action_id ) {
-		self::process_categories( $manifest['categories'], $logger_action_id );
+		$doc_ids = array();
+		self::process_categories( $manifest['categories'], $logger_action_id, $doc_ids );
+		return $doc_ids;
```

Comment:
In my [open PR](https://github.com/woocommerce/woocommerce/pull/39395/files#diff-d4caf3f44486acab0cda29402f3af82b5b796a1a921bce003c1db2767e7a4632) I created a function to collect `$doc_ids` from a manifest, `collect_doc_ids_from_manifest`. To have `process_categories` perform the same function as a side effect of creating and updating posts is convenient here, but we will have the same logic repeated. If I merge mine after, I can clean this up when dealing with conflicts.

Otherwise, you could c/p the function and its unit tests. What do you think?

Reply:
Happy to bring across your function and tests here :) I'll work on that.

Reply:
@psealock I had a look at your PR and I think actually conflicts won't be a lot and it makes more sense just to fix it up in your PR afterwards if thats ok? I started moving things partially across and it just felt like.. messy since your collect method is in a separate class.